tsdb/series_file.go

```diff
@@ -445,8 +448,11 @@ func CompareSeriesKeys(a, b []byte) int {
 	}
 
 	// Read total size.
-	_, a = ReadSeriesKeyLen(a)
-	_, b = ReadSeriesKeyLen(b)
+	szA, a := ReadSeriesKeyLen(a)
+	szB, b := ReadSeriesKeyLen(b)
+	if szA == 0 && szB == 0 {
+		return 0
+	}
```

Comment:
I believe you still need to fix this test up, to avoid calls to `ReadSeriesKeyMeasurement` with empty buffers.  Either the way I suggested or the way you suggested from the start of the function, or some combination, but I would like to see the operation halted if either key is zero length.

Reply:
I believe I've overcomplicated this when I was first reading it, those initial checks are okay and we can conduct something very similar because the result for our `sz` is only ever `0` on an error, which is bubbled up from [`binary.Uvarint`](https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/encoding/binary/varint.go;l=69-88):
```go
func Uvarint(buf []byte) (uint64, int) {
	var x uint64
	var s uint
	for i, b := range buf {
		if i == MaxVarintLen64 {
			// Catch byte reads past MaxVarintLen64.
			// See issue https://golang.org/issues/41185
			return 0, -(i + 1) // overflow
		}
		if b < 0x80 {
			if i == MaxVarintLen64-1 && b > 1 {
				return 0, -(i + 1) // overflow
			}
			return x | uint64(b)<<s, i + 1
		}
		x |= uint64(b&0x7f) << s
		s += 7
	}
	return 0, 0
}
```

I've pulled similar checks further down to the `ReadSeriesKeyLen` as well to cover it, it looks like similar logic was done on a tags check further down here too.